### PR TITLE
dockershim: check if f.Sync() returns an error and surface it

### PR DESCRIPTION
**Pull request by yujuhong**
```release-note
dockershim: check the error when syncing the checkpoint.
```

### CODE REVIEW

File: pkg/kubelet/dockershim/checkpoint_store.go [hunk 1]
```diff
@@ -76,7 +76,10 @@ func writeFileAndSync(filename string, data []byte, perm os.FileMode) error {
 	if err == nil && n < len(data) {
 		err = io.ErrShortWrite
 	}
-	f.Sync()
+	if err == nil {
+		// Only sync if the Write completed successfully.
+		err = f.Sync()
```
Comment (mtaufen) [inline pkg/kubelet/dockershim/checkpoint_store.go]: Is there a reason for hiding old errors behind new errors? If it's so control flow makes it to `f.Close()`, perhaps `defer` is better?

Comment (yujuhong) [reply pkg/kubelet/dockershim/checkpoint_store.go]: This function surfaces the errors in the following order:
 1. write error
 2. sync error
 3. close error

If (1) is non-nil, return (1). If (2) is non-nil, return (2), etc.
This follows the existing flow and convention in `ioutil.WriteFile()`

Comment (mtaufen) [reply pkg/kubelet/dockershim/checkpoint_store.go]: Ah, I see. SG.

**Pull request merged**